src/Console/Command/CompareCommand.php

```diff
@@ -0,0 +1,323 @@
+
+        /**
+         * @var string
+         * @var RuleSet $set
+         */
+        foreach ($this->ruleSets as $name => $set) {
+            /**
+             * @var string
+             * @var bool   $value
+             */
+            foreach ($set->getRules() as $rule => $value) {
+                if (false === isset($this->list[$rule]['in_set'])) {
+                    $this->list[$rule]['in_set'] = [];
+                }
+                $this->list[$rule]['in_set'][] = $name;
+            }
+        }
+
+        /** @var FixerInterface $fixer */
+        foreach ($this->builtInFixers as $fixer) {
+            if ($isInherited = !empty($this->list[$fixer->getName()]['in_set'])) {
+                ++$this->countInherited;
+            }
+
+            $this->list[$fixer->getName()]['name'] = $fixer->getName();
+            $this->list[$fixer->getName()]['is_configured'] = $this->isFixerConfigured($fixer);
+            $this->list[$fixer->getName()]['is_enabled'] = $this->isFixerEnabled($fixer);
+            $this->list[$fixer->getName()]['is_risky'] = $this->isFixerRisky($fixer);
+            $this->list[$fixer->getName()]['is_inherited'] = $isInherited;
+
+            if ($this->isFixerRisky($fixer)) {
+                $this->riskyFixers[] = $fixer->getName();
+            }
+        }
+
+        // Render the table
+        $this->buildTable($output)->render();
+
+        if ($input->getOption('dump')) {
+            $this->dump($output);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $resolver = new ConfigurationResolver(
+            $this->defaultConfig,
+            [
+                'allow-risky' => true,
+                'config' => $input->getOption('config'),
+            ],
+            getcwd()
+        );
+
+        $this->builtInFixers = $this->fixerFactory->getFixers();
+        $this->configuredFixers = $resolver->getConfig()->getRules();
+        $this->enabledFixers = $resolver->getRules();
+
+        // Get the RuleSets and their Fixers
+        foreach (RuleSet::create()->getSetDefinitionNames() as $setName) {
+            $this->ruleSets[$setName] = new RuleSet([$setName => true]);
+        }
+
+        // Order alphabetically
+        usort($this->builtInFixers, function (FixerInterface $a, FixerInterface $b) {
+            return strcmp($a->getName(), $b->getName());
+        });
+
+        $this->hideConfigured = $input->getOption('hide-configured');
+        $this->hideEnabled = $input->getOption('hide-enabled');
+        $this->hideRisky = $input->getOption('hide-risky');
+        $this->hideInherited = $input->getOption('hide-inherited');
+    }
+
+    /**
+     * @param OutputInterface $output
+     *
+     * @return Table
+     */
+    private function buildTable(OutputInterface $output)
+    {
+        $table = new Table($output);
+
+        $columns = [
```

Comment:
The table is starting to get quite large. Maybe we could reduce the number of columns.

I would suggest to replace _Configured_ and _Enabled_ (and maybe _Inherited_ too?) columns with a simple symbol before the fixer name, e.g.:
```
✓ fixer_enabled_explicitly
✖ fixer_not_enabled
? fixer_enabled_implicitly_by_a_ruleset
```
(no idea what would be the symbol for the third case).

WDYT?

Reply:
Yes, it is going to be quite large.

The problem is that the symbols are UNICODE and there are not a lot of them.

Here is a list of the available one: http://www.utf8-chartable.de/unicode-utf8-table.pl?start=9984&number=128&names=-&utf8=string-literal

Can you find one that is suitable to represent an inherited `Fixer`?

In the mean time a solution to consider is to put the inheritance path directly under the Fixer name (as is the path that makes the table really large)...

I'll make some variations to see how it will look like...

Reply:
Maybe we can always use `✓` when the fixer is enabled, and choose a different color depending on how it is enabled, e.g. green for explicitly configured fixers and yellow for inherited ones?

We could also replace symbols with actual words when verbosity level is increased? This would allow color blind people to still be able to get all the details.

Reply:
Yes, I've done exactly this: green for Fixers configured in `.php_cs`, yellow for inherited, red for not enabled.

See the comment I'm going to write (be patient! :) )